### PR TITLE
Add support for clippy.toml

### PR DESCRIPTION
**Pull request by djmarcin**
This allows users to override clippy settings via the usual config file in their workspace.

### CODE REVIEW

File: rust/private/clippy.bzl [hunk 1]
```diff
@@ -116,9 +116,14 @@ def _clippy_aspect_impl(target, ctx):
     if crate_info.is_test:
         args.add("--test")
 
+    valid_config_file_names = [".clippy.toml", "clippy.toml"]
```
Comment (UebelAndre) [inline rust/private/clippy.bzl:119]: Does clippy care about the file name? Why can't this be arbitrary?

File: tools/clippy/BUILD [hunk 2]
```diff
@@ -0,0 +1 @@
+exports_files(["clippy.toml"])
```
Comment (UebelAndre) [inline tools/clippy/BUILD:1]: Can you rename this file `BUILD` file to `BUILD.bazel` for consistency with the rest of the repo?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Nice! Just a few questions but a very welcomed change! 😄

File: rust/private/clippy.bzl (hunk 1 above)
Comment (djmarcin) [reply rust/private/clippy.bzl:119]: Yes, clippy cares about the filename. https://github.com/rust-lang/rust-clippy/blob/cadb93b6de1b5ba5aca53c5c4c7987f3c09962a8/clippy_lints/src/utils/conf.rs#L220-L222

File: tools/clippy/BUILD (hunk 2 above)
Comment (djmarcin) [reply tools/clippy/BUILD]: Renamed

**Review by UebelAndre — CHANGES_REQUESTED** [review]
It looks like this was added in https://github.com/rust-lang/rust-clippy/commit/86c513e605e7982c00587e05a33a1062ca9a5334 which was first released with Rust `1.34.0`. Maybe that's something we should add to the documentation?

**Comment by UebelAndre** [comment]
Additionally, can you also add a comment at the top of the default `clippy.toml`
```toml
# clippy options: https://rust-lang.github.io/rust-clippy/
```

Similar to the [rustfmt.toml](https://github.com/bazelbuild/rules_rust/blob/86a653f32535f75f0dedf432cdbe9309739e4db1/tools/rustfmt/rustfmt.toml) equivalent?

**Comment by djmarcin** [comment]
> Additionally, can you also add a comment at the top of the default `clippy.toml`
> 
> ```toml
> # clippy options: https://rust-lang.github.io/rust-clippy/
> ```
> 
> Similar to the [rustfmt.toml](https://github.com/bazelbuild/rules_rust/blob/86a653f32535f75f0dedf432cdbe9309739e4db1/tools/rustfmt/rustfmt.toml) equivalent?

Done.

File: rust/private/clippy.bzl [hunk 3]
```diff
@@ -116,6 +116,12 @@ def _clippy_aspect_impl(target, ctx):
...
+    if ctx.file._config.basename not in valid_config_file_names:
+        fail("The clippy config file must be named one of: %s" % valid_config_file_names)
+    env["CLIPPY_CONF_DIR"] = "${pwd}/%s" % ctx.file._config.dirname
```
Comment (UebelAndre) [inline rust/private/clippy.bzl:122]:
```suggestion
    env["CLIPPY_CONF_DIR"] = "${pwd}/{}".format(ctx.file._config.dirname)
```

File: rust/private/clippy.bzl [hunk 4]
```diff
@@ -116,6 +116,12 @@ def _clippy_aspect_impl(target, ctx):
     if crate_info.is_test:
         args.add("--test")
 
+    valid_config_file_names = [".clippy.toml", "clippy.toml"]
+    if ctx.file._config.basename not in valid_config_file_names:
+        fail("The clippy config file must be named one of: %s" % valid_config_file_names)
```
Comment (UebelAndre) [inline rust/private/clippy.bzl:121]:
```suggestion
        fail("The clippy config file must be named one of: {}".format(valid_config_file_names))
```

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Two small nits and, while this probably isn't a fantastic example, could you also setup something similar to [//test/rustfmt/rustfmt_failure_test](https://github.com/bazelbuild/rules_rust/blob/86a653f32535f75f0dedf432cdbe9309739e4db1/test/rustfmt/rustfmt_failure_test.sh#L79) which actually uses the config? I realize the `rustfmt` case is kinda bad because it doesn't enforce that the particular setting is actually used, but at least the starlark code path would be exercised in a workspace that's not `rules_rust`.

File: rust/private/clippy.bzl (hunk 1 above)
Comment (UebelAndre) [reply rust/private/clippy.bzl:119]:
```suggestion
    # Clippy itself looks for specifically named files so the same
    # enforcement is applied here.
    valid_config_file_names = [".clippy.toml", "clippy.toml"]
```

File: test/clippy/too_many_args/clippy.toml [hunk 5]
```diff
@@ -0,0 +1 @@
+too-many-arguments-threshold = 2
```
Comment (UebelAndre) [inline test/clippy/too_many_args/clippy.toml:1]: nit: can you add a newline here?

**Review by UebelAndre — APPROVED** [review]
Awesome! Some tiny nits if you don't mind but otherwise LGTM!

**Review by UebelAndre — APPROVED** [review]

**Pull request merged**